tests/Unit/ProxyClient/AbstractProxyClientTest.php

```diff
@@ -0,0 +1,282 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCache package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCache\Tests\Unit\ProxyClient;
+
+use FOS\HttpCache\Exception\ExceptionCollection;
+use FOS\HttpCache\ProxyClient\Varnish;
+use Guzzle\Http\Client;
+use Guzzle\Http\Exception\CurlException;
+use Guzzle\Http\Exception\MultiTransferException;
+use Guzzle\Http\Exception\RequestException;
+use Guzzle\Plugin\Mock\MockPlugin;
+use Guzzle\Http\Message\Response;
+use Guzzle\Http\Message\Request;
+use \Mockery;
+
+/**
+ * Testing the base methods of the proxy client, using the Varnish client as concrete class.
```

Comment:
yeah. but its rather complicated and duplicating the tests that are run. tests don't take that long so it would not hurt. want to do it? otherwise i propose we leave as is for now.